### ISSUE TITLE
Cannot choose a device to record

### ISSUE THREAD

**Issue by HydroCarbon** [llfbandit/record#296]
**Package version**

**Environment**
 - OS: macOS 14.3.1
 - Browser: Chrome

**Describe the bug**

I use this config to record stream:

```dart
RecordConfig config = RecordConfig(
      device: _inputDevice,
      encoder: AudioEncoder.pcm16bits,
      sampleRate: 16000,
      numChannels: 1,
);

Stream<Uint8List> recordStream = await _audioRecorder.startStream(config);
```

Expected to record audio from choosen device(`_inputDevice`), and this is code to choose device:

```dart
_audioRecorder
        .listInputDevices()
        .then((devices) => devices.firstWhere(
            (device) => device.label == deviceLabel,
            orElse: () => devices.first))
        .then((device) => _inputDevice = device)
```

when I use diffenrent label, it always record from macos's microphone device.
**By the way, i use Blackhole create a virtual input device.**

**To Reproduce**

1. Create a virtual device use Blackhole[https://github.com/ExistentialAudio/BlackHole](https://github.com/ExistentialAudio/BlackHole)
2. Choose virtual input device created by Blackhole to record stream.
3. Still record from default macos's microphone input device.

**Expected behavior**

Record from device that i choosen

**Additional context**
There is blackhole's url [https://github.com/ExistentialAudio/BlackHole](https://github.com/ExistentialAudio/BlackHole)

**Comment by HydroCarbon** [comment]
And I found it always use the input device that setted in system setting.
<img width="715" alt="image" src="https://github.com/llfbandit/record/assets/30232379/418f47d2-62d4-48a2-8414-9a7057a68fc8">

**Comment by austibwu** [comment]
@llfbandit is device selection an intended feature of the plugin?

**Comment by HydroCarbon** [comment]
> @llfbandit is device selection an intended feature of the plugin?

Record has a param of device.I think it's a param to select device.
And document say it can choose device on macOS.
![截图 2024-04-01 17-16-06](https://github.com/llfbandit/record/assets/30232379/a1552f52-6b1e-47fa-af88-fa55897ee2d6)

**Comment by llfbandit** [comment]
@HydroCarbon 
Device selection was missing on streaming code branch.

Are you able to test it from this repository on master branch?

**Comment by HydroCarbon** [comment]
@llfbandit 
OK，I will test it in two days. and I will reply to you .

**Comment by llfbandit** [comment]
Released in record_darwin 1.1.0
